File: next-greater-element-i/next-greater-element-i.cs
public class Solution {
    public int[] NextGreaterElement(int[] nums1, int[] nums2) {
        
        Stack<int> stack=new Stack<int>();
        
        Dictionary<int,int> map=new Dictionary<int,int>();
        
        for(int i=0;i<nums2.Length;i++)
        {
            while(stack.Count>0 && nums2[i]>nums2[stack.Peek()])
            {
                //Console.WriteLine(nums2[i]+","+stack.Peek());
                int j=stack.Pop();
                map.Add(nums2[j],nums2[i]);
            }
            
            stack.Push(i);
        }
        
        int[] res=new int[nums1.Length];
        for(int i=0;i<nums1.Length;i++)
        {
            if(map.ContainsKey(nums1[i]))
            {
                res[i]=map[nums1[i]];
            }
            else
                res[i]=-1;
        }
        
        return res;
    }
}

File: next-greater-element-i/next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        
        Stack<Integer> monoDecStack=new Stack<Integer>();
        
        //int[] reff=new int[nums2.length];
        
        HashMap<Integer,Integer> map=new HashMap<>();
        
       // Arrays.fill(reff,-1);
        
        for(int i=0;i<nums2.length;i++)
        {
            while(monoDecStack.size()>0 && nums2[monoDecStack.peek()]<nums2[i])
            {
                //reff[monoDecStack.pop()]=nums2[i];
                map.put(nums2[monoDecStack.pop()],nums2[i]);
            }
            
            monoDecStack.push(i);
        }
        
        
        
        for(int i=0;i<nums1.length;i++)
        {
            if(!map.containsKey(nums1[i]))
            {
                nums1[i]=-1;
            }
            else
                nums1[i]=map.get(nums1[i]);
        }
        
        return nums1;
    }
}